firmware.S

```diff
@@ -12,6 +12,13 @@
 #define SEC_PROXY_STATUS		0x2a380000
 #define SEC_PROXY_MSG_DATA		0x2a480004
 
+#define PON_REASON_SOF_ADDR     	0xa2200000
+#define PON_REASON_MAGIC_ADDR   	0xa2200008
+#define PON_REASON_EOF_ADDR     	0xa2200010
+#define PON_REASON_SOF_NUM      	0xBBBBCCCC
+#define PON_REASON_MAGIC_NUM    	0xDDDDDDDD
+#define PON_REASON_EOF_NUM      	0xCCCCBBBB
```

Comment:
Why do we need so many values and magics if we reserve/define the address where the reset reason is written to? The firmware must be started anyway and could initialize the memory cell also after cold boots.

Reply:
I was thinking to add the reliability. Anyway I am also okay to simplify it. Do you prefer to simplify it?

> The firmware must be started anyway and could initialize the memory cell also after cold boots.  

I am not sure whether I understand this correctly.  If you were suggesting that the data is not safe for cold boots, that would not be a problem, because this patch only tries to save the non-volatile data for warm boots.

Reply:
That suggestion made no sense: We can't get the reset reason RAM into a well-defined state after cold-boot or hard-reset as we can't tell that apart from watchdog-triggered reboots.

Anyway, we need a proper definition of the interface first. Not here, best in the DT bindings that need to be written first. Part of that definition needs to be who is supposed to wipe the pattern again, and when.

Reply:
If I understand correctly, you were suggesting that I should first define the proper interface and discuss it with the community, and then design the entire process based on the conclusion. 

However, this is kind of my plan, I will leave this until I get a conclusion.

Reply:
Then we are on the same page. Once the interface is settle, this here can be merged (or first adjusted).

BTW, I would then suggest to make the base address of the magic values build-time configurable (make variable), maybe with a default we will also use.

Reply:
Hi @jan-kiszka ,

The Linux community has accepted my patches, although I don't know when they will be merged. Our project now requires this functionality, so are we going to continue with the current PR progress?

Reply:
Did you get any notification that at least the DT bindings are in some staging tree, lining up for the next merge window (6.6)? Do you have an ack by the DT bindings maintainer as well already?

The reason I'm a bit picky here is that we are creating an ABI between firmware and kernel, and both pieces are usually not updated in lock-step. We would run into troubles if we have to bump the firmware in the field just because we move to a mainline kernel.

Reply:
The v6 patches were reviewed-by Krzysztof and Guenter on July 18.  I was suggested to not  re-post patches and the upstream maintainer will do the following job on the version they apply.